controllers/clusteroperator.go

```diff
@@ -149,3 +149,24 @@ func (r *ProvisioningReconciler) updateCOStatusDisabled() error {
 
 	return r.syncStatus(co, conds)
 }
+
+// updateCOStatusDegraded updates the ClusterOperator's Degraded
+// degradedReason should contain the current reason for the Operator to be marked in that state
+func (r *ProvisioningReconciler) updateCOStatusDegraded(degradedReason string, detailedError string) error {
+	degradedMessage := "Operator is Degraded"
+	progressingMessage := "Operator is Degraded while Progressing"
+
+	co, err := r.getOrCreateClusterOperator()
+	if err != nil {
+		r.Log.Error(err, "failed to get or create ClusterOperator")
```

Comment:
Is logging here before returning the error going to mean the error is logged multiple times? Should this just wrap the error with more detail and then return it?

Reply:
Removed this error msg. getOrCreateClusterOperator() is returning a good error message that can be wrapped with one  from Reconcile() and returned from there. Doing the same for updateCOStatusDisabled().